backend/compact-connect/stacks/api_stack/v1_api/credentials.py

```diff
@@ -0,0 +1,121 @@
+from __future__ import annotations
+
+import os
+
+from aws_cdk import Duration
+from aws_cdk.aws_apigateway import LambdaIntegration, MethodOptions, MethodResponse, Resource
+from aws_cdk.aws_iam import Effect, PolicyStatement
+from cdk_nag import NagSuppressions
+from common_constructs.python_function import PythonFunction
+from common_constructs.stack import Stack
+
+# Importing module level to allow lazy loading for typing
+from stacks.api_stack import cc_api
+
+from .api_model import ApiModel
+
+
+class Credentials:
+    def __init__(
+        self,
+        *,
+        resource: Resource,
+        method_options: MethodOptions,
+        api_model: ApiModel,
+    ):
+        super().__init__()
+
+        self.resource = resource
+        self.api: cc_api.CCApi = resource.api
+        self.api_model = api_model
+
+        stack: Stack = Stack.of(resource)
+        lambda_environment = {
+            **stack.common_env_vars,
+        }
+
+        # /v1/compacts/{compact}/credentials/payment-processor
+        self._add_post_credentials_payment_processor(
+            method_options=method_options,
+            lambda_environment=lambda_environment,
+        )
+
+    def _add_post_credentials_payment_processor(
+        self,
+        method_options: MethodOptions,
+        lambda_environment: dict,
+    ):
+        self.payment_processor_resource = self.resource.add_resource('payment-processor')
+
+        self.post_credentials_payment_processor_handler = self._post_credentials_payment_processor_handler(
+            lambda_environment=lambda_environment,
+        )
+        self.api.log_groups.append(self.post_credentials_payment_processor_handler.log_group)
+
+        self.payment_processor_resource.add_method(
+            'POST',
+            request_validator=self.api.parameter_body_validator,
+            request_models={'application/json': self.api_model.post_credentials_payment_processor_request_model},
+            method_responses=[
+                MethodResponse(
+                    status_code='200',
+                    response_models={
+                        'application/json': self.api_model.post_credentials_payment_processor_response_model
+                    },
+                ),
+            ],
+            integration=LambdaIntegration(
+                self.post_credentials_payment_processor_handler, timeout=Duration.seconds(29)
```

Comment:
Why 29, vs 30?

Reply:
API Gateway has a timeout limit of 30 seconds, so if the lambda is unable to finish its task in that time window we want to know it was the lambda that timed out and not API Gateway.

Reply:
Got it. Mind adding a brief comment?

Reply:
done